modules/chemical_reactions/src/auxkernels/ThermochimicaAux.C

```diff
@@ -0,0 +1,108 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThermochimicaAux.h"
+#include "libmesh/int_range.h"
+
+#include <iostream>
+#include <string>
+#include <cmath>
+
+#ifdef THERMOCHIMICA_ENABLED
+#include "Thermochimica-cxx.h"
+
+registerMooseObject("ChemicalReactionsApp", ThermochimicaAux);
+#endif
+
+InputParameters
+ThermochimicaAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addRequiredCoupledVar("elements", "Amounts of elements");
+  params.addCoupledVar("output_phases", "Amounts of phases to be output");
+  params.addCoupledVar("output_species", "Amounts of species to be output");
+  params.addRequiredParam<UserObjectName>("thermo_nodal_data_uo", "Name of the user object");
+  params.addCoupledVar("element_potentials", "Chemical potentials of elements");
+
+  params.addClassDescription("Extracts phase and species amounts, and element chemical potentials "
+                             "from a Thermochimica user object.");
```

Comment:
ThermochimicaNodalData UO ?

Reply:
maybe the def of 'amounts' should be here instead of each param

Reply:
There will be derived UOs in the future